cpp/src/arrow/acero/query_context.cc

```diff
@@ -23,6 +23,36 @@ namespace arrow {
 using arrow::internal::CpuInfo;
 namespace acero {
 
+namespace internal {
+
+int64_t GetTempStackSizeFromEnvVar() {
+  auto maybe_env_value = arrow::internal::GetEnvVar(kTempStackSizeEnvVar);
+  if (!maybe_env_value.ok()) {
+    return kDefaultTempStackSize;
+  }
+  auto env_value = *std::move(maybe_env_value);
+  if (env_value.empty()) {
+    return kDefaultTempStackSize;
+  }
+
+  int64_t temp_stack_size = 0;
+  size_t length = 0;
+  bool exception = false;
+  try {
+    temp_stack_size = std::stoll(env_value.c_str(), &length);
+  } catch (const std::exception&) {
+    exception = true;
+  }
+  if (length != env_value.length() || exception || temp_stack_size <= 0) {
```

Comment:
The minimum valid `temp_stack_size` should be greater than `0` to prevent someone that can control env vars from forcing a very small temp stack size.

Reply:
I'm saying you should not allocate very small temp stacks. Give them at least 64 kilobytes even if users set a lower value.

Reply:
The default today is 256K, so a minimum of 64K seems reasonable.

64K is allocation granularity on Windows [1] and is a very common default in Arrow (for other things) [2].

[1] https://devblogs.microsoft.com/oldnewthing/20031008-00/?p=42223
[2] https://github.com/search?q=repo%3Aapache%2Farrow%2064k&type=code

Reply:
Thank you for the info. 64K seems perfect! Will update soon.

Reply:
Addressed.